File: pingdart-python/pingdart_sdk/services/calls_service.py
class CallsService:
    def __init__(self, http_client):
        self.http = http_client

    def list_apps(self):
        """
        List all call applications for the authenticated user.
        """
        try:
            response = self.http.get("/v1/calls/apps")
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to list call apps: {str(e)}")

    def create_app(self, name, app_type='Web'):
        """
        Create a new call application.
        """
        try:
            response = self.http.post("/v1/calls/apps", json={"name": name, "type": app_type})
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to create call app: {str(e)}")

    def delete_app(self, app_id):
        """
        Delete a call application.
        """
        try:
            response = self.http.delete(f"/v1/calls/apps/{app_id}")
            response.raise_for_status()
            return response.json()
        except Exception as e:
            raise Exception(f"Failed to delete call app: {str(e)}")

File: pingdart-node/src/services/calls.service.js
class CallsService {
    constructor(http, config) {
        this.http = http;
    }

    /**
     * List all call applications for the authenticated user.
     * @returns {Promise<Object>} The server response containing the list of apps.
     */
    async listApps() {
        try {
            const response = await this.http.get("v1/calls/apps");
            return response.data;
        } catch (error) {
            throw new Error(`Failed to list call apps: ${error.message}`);
        }
    }

    /**
     * Create a new call application.
     * @param {string} name - The name of the application.
     * @param {string} [type='Web'] - The type of application (e.g., Web, Mobile).
     * @returns {Promise<Object>} The server response containing the created app details.
     */
    async createApp(name, type = 'Web') {
        try {
            const response = await this.http.post("v1/calls/apps", { name, type });
            return response.data;
        } catch (error) {
            throw new Error(`Failed to create call app: ${error.message}`);
        }
    }

    /**
     * Delete a call application.
     * @param {number|string} id - The ID of the application to delete.
     * @returns {Promise<Object>} The server response.
     */
    async deleteApp(id) {
        try {
            const response = await this.http.delete(`v1/calls/apps/${id}`);
            return response.data;
        } catch (error) {
            throw new Error(`Failed to delete call app: ${error.message}`);
        }
    }
}

export default CallsService;
